Documentation/upgrade.md

```diff
@@ -0,0 +1,293 @@
+It is possible while upgrading your monitor pods that the operator will find them out of quorum and immediately replace them with a new monitor, such as `mon0` getting replaced by `mon3`.
+This is okay as long as the cluster health looks good and all monitors eventually reach quorum again.
+
+### Object Storage Daemons (OSDs)
+The OSD pods can be managed in two different ways, depending on how you specified your storage configuration in your [Cluster spec](./cluster-crd.md#cluster-settings).  
+* **Use all nodes:** all storage nodes in the cluster will be managed by a single daemon set.
+Only the one daemon set will need to be edited to update the image version, then each OSD pod will need to be deleted so that a new pod will be created by the daemon set to take its place.
+* **Specify individual nodes:** each storage node specified in the cluster spec will be managed by its own individual replica set.
+Each of these replica sets will need to be edited to update the image version, then each OSD pod will need to be deleted so its replica set will start a new pod on the new version to replace it.
+
+In this example, we are going to walk through the case where `useAllNodes: true` was set in the cluster spec, so there will be a single daemon set managing all the OSD pods.
+
+Let's edit the image version of the pod template in the daemon set:
+```bash
+kubectl -n rook edit daemonset rook-ceph-osd
+```
+
+The relevant snippet to update in your editor is shown updated to `rook/rook:v0.5.1` below:
+```
+        image: rook/rook:v0.5.1
+        imagePullPolicy: IfNotPresent
+        name: rook-ceph-osd
+```
+
+Once the daemon set is updated, we can begin deleting each OSD pod **one at a time** and verifying a new one comes up to replace it that is running the new version.
+After each pod, the cluster health and OSD status should remain or return to an okay state as described in the [health verification section](#health-verification).
+To get the names of all the OSD pods, the following can be used:
+```bash
+kubectl -n rook get pod -l app=rook-ceph-osd -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}'
+```
+
+Below is an example of deleting just one of the OSD pods (note that the names of your OSD pods will be different):
+```bash
+kubectl -n rook delete pod rook-ceph-osd-kcj8f
+```
+
+The status and version for all OSD pods can be collected with the following command and its sample output:
+```bash
+> kubectl -n rook get pod -l app=rook-ceph-osd -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.phase}{" "}{.spec.containers[0].image}{"\n"}{end}'
+rook-ceph-osd-7fmdl Running rook/rook:v0.5.1
+rook-ceph-osd-ntfx0 Running rook/rook:v0.5.1
+```
+
+Remember after each OSD pod to verify the cluster health using the instructions found in the [health verification section](#health-verification).
+
+### Ceph Manager
```

Comment:
There is a lot of similarity between the different daemons. You might consider a common section, with the specific sections about each of the daemons could be simpler. Then rgw and mds would be simple additions.

Reply:
i'll see if there's some refactoring i can do.  i have a feeling that @kokhang suggestion of using the `kubectl set image` command instead of `kubectl edit` might be sufficient on its own to slim things down.  I'll see.